slave-setup/worker_installer/slave_setup.sh

```diff
@@ -0,0 +1,254 @@
+sudo apt-get -y install dbus-x11
+
+#mkdocs configuration
+# Installing python-pip
+echo ""
+echo ""
+echo "Installing pip..............................................,"
+echo ""
+echo ""
+sudo apt-get -y install python-pip
+echo ""
+echo ""
+sudo pip --version
+echo ""
+echo ""
+# Installing build-essential
+echo ""
+echo ""
+echo "Installing build-essential..................................,"
+echo ""
+echo ""
+sudo apt-get -y install build-essential python-dev
+echo ""
+echo ""
+# Installing mkdocs
+echo ""
+echo ""
+echo "Installing mkdocs..............................................,"
+echo ""
+echo ""
+sudo pip install mkdocs && mkdocs --version
+echo ""
+echo ""
+# Installing mkdocs-material
+echo ""
+echo ""
+echo "Installing mkdocs-material....................................,"
+echo ""
+echo ""
+sudo pip install mkdocs-material
+echo ""
+echo ""
+
+
+#Creating Directories
+cd /build/jenkins-home
+mkdir -p software/java
+mkdir -p software/maven
+mkdir -p software/jce
+mkdir -p software/nodejs
+cd
+
+# defining JDK, MAVEN, JCE and NODEJS variables. please change here when required to add JDK/MAVEN and etc..
+JDK7x=jdk-7u51-linux-x64.tar.gz
+JDK8x=jdk-8u45-linux-x64.tar.gz
+JDK81x=jdk-8u144-linux-x64.tar.gz
+
+APACHE_MAVEN_22x=apache-maven-2.2.1-bin.tar.gz
+APACHE_MAVEN_30x=apache-maven-3.0.5-bin.tar.gz
+APACHE_MAVEN_31x=apache-maven-3.1.1-bin.zip
+APACHE_MAVEN_32x=apache-maven-3.2.2-bin.tar.gz
+APACHE_MAVEN_33x=apache-maven-3.3.9-bin.zip
+
+JCE7=UnlimitedJCEPolicyJDK7.zip
+JCE8=jce_policy-8.zip
+
+NODEJSv6=node-v6.10.0-linux-x64.tar.xz
+
+# unzip installation file
+#unzip java files
+echo ""
+echo "*****************************************************************************"
+echo "*                            Extracting Java files                          *"
+echo "*****************************************************************************"
+echo ""
+
+tar -zxvf /build/jenkins-home/slaveSetupFile/$JDK7x -C /build/jenkins-home/software/java
+#tar -zxvf /build/jenkins-home/slaveSetupFile/jdk-8u45-linux-x64.tar.gz -C /build/jenkins-home/software/java
+tar -zxvf /build/jenkins-home/slaveSetupFile/$JDK81x -C /build/jenkins-home/software/java
+
+
+#unzip maven
+echo ""
+echo "*****************************************************************************"
+echo "*                           Extracting Maven files                          *"
+echo "*****************************************************************************"
+echo ""
+tar -zxvf /build/jenkins-home/slaveSetupFile/$APACHE_MAVEN_22x -C /build/jenkins-home/software/maven
+tar -zxvf /build/jenkins-home/slaveSetupFile/$APACHE_MAVEN_30x -C /build/jenkins-home/software/maven
+tar -zxvf /build/jenkins-home/slaveSetupFile/$APACHE_MAVEN_32x -C /build/jenkins-home/software/maven
+
+unzip -o /build/jenkins-home/slaveSetupFile/$APACHE_MAVEN_31x -d /build/jenkins-home/software/maven
+unzip -o /build/jenkins-home/slaveSetupFile/$APACHE_MAVEN_33x -d /build/jenkins-home/software/maven
```

Comment:
We should download these files from apache site. The download links are available here - https://archive.apache.org/dist/maven/binaries/